src/types/theme.ts

```diff
@@ -92,3 +92,7 @@ interface ITheme {
   };
 }
 export default ITheme;
+
+export interface StyledProps {
+  theme: ITheme;
+}
```

Comment:
I believe it'd work the same way, you don't need to declare massive `ITheme`

```suggestion
export interface StyledProps {
  theme: typeof theme;
}
```